### PR TITLE
Made members that store sensor data properties. Removed deprecated members. disabled members return none

### LINKED ISSUE #37: sensor.euler,quaternion,gyro, etc. returns 0 instead of None when those sensors are disabled
I think that to avoid confusion, the module should return something that isn't 0 so people know when the sensors are turned off since a sensor.whatever could _technically_ return 0 but still be active.

### PR DESCRIPTION
**Pull request by dherrada**
It's really late, and I haven't tested everything about this, but I don't anticipate anything major about the structure of the code changing if I find a little bug or two when I'm testing it. Fixes #37

### CODE REVIEW

**Comment by dherrada** [comment]
Alright. Just tested this and everything seems to be working fine

File: adafruit_bno055.py [hunk 1]
```diff
@@ -244,3 +222,58 @@ def use_external_crystal(self, value):
...
+    def temperature(self):
+        """Measures the temperature of the chip in degrees Celsius."""
+        return self._temperature
+
+    @property
+    def acceleration(self):
+        """Gives the raw accelerometer readings, in m/s."""
+        if self.mode not in [0, 2, 3, 6]:
+            return self._acceleration
+        return (None, None, None)
```
Comment (siddacious) [inline adafruit_bno055.py]: I would also update the docstrings of the mode property and the new properties that may return `None` to describe the new behavior.

**Comment by dherrada** [comment]
@siddacious Does this look good? I wasn't really sure what to do about the mode property's docstring.

File: adafruit_bno055.py (hunk 1 above)
Comment (siddacious) [reply adafruit_bno055.py]: The docstrings in the individual properties are great. All I would do for the mode property is to add a legend at the top of the table to describe what each symbol in the table means  (x=on, -=off and will return an empty tuple)

**Comment by siddacious** [comment]
So close! Just fix the whitespace issue that pylint is complaining about and we're good to go

**Comment by caternuson** [comment]
Oops. Too late. FWIW - the lists could've been tuples. Ex:
```python
if self.mode not in (0, 2, 3, 6):
```

**Comment by dherrada** [comment]
I’ll make sure to change that before I make the next release for this library

**Pull request merged**